### PR TITLE
Added new eslint-rule regular-expression-comment

### LINKED ISSUE #33: Add require-regular-expression-comment rule
Required to address https://github.com/checkdigit/eslint-config/issues/113

### PR DESCRIPTION
**Pull request by jpolavar**
Closes #33

### CODE REVIEW

File: src/regular-expression-comment.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  },
+  create(context) {
+    return {
+      Literal(node) {
+        if (node.value instanceof RegExp) {
+          const regexLine = node.loc?.start.line;
+          if (regexLine) {
+            const previousLine = regexLine - 1;
+            const hasRegexComment =
+              context.sourceCode.getAllComments().some((comment) => comment.loc?.start.line === previousLine) ||
```
Comment (carlansley) [inline src/regular-expression-comment.ts:27]: this is a little inefficient, to start with we only should scan all comments once (e.g. `... .some((comment) => comment.loc?.start.line === previousLine || comment.loc?.start.line === regexLine)`).  even then seems excessive, but I suppose there aren't normally a ton of regexs in our code.  Is there a more efficient way to see if there is a comment on this line or the previous, other than scanning a list of all comments in the code?

**Review by carlansley — CHANGES_REQUESTED** [review]
Potential efficiency concern, has this been tried out on an internal codebase with a bunch of regexes?  (not sure what that would be, @ramaghanta might know)

**Comment by carlansley** [comment]
(adding @bobwieler since this was originally his issue)

File: src/regular-expression-comment.ts (hunk 1 above)
Comment (jpolavar) [reply src/regular-expression-comment.ts:27]: Sure  we do have something like getCommentsBefore(node) that would check before this line with regex similarly getCommentsInside(node). But it somehow doesn't return the comments even though they exist, so have to rewrite this way.

**Review by carlansley — APPROVED** [review]

File: src/regular-expression-comment.spec.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * Copyright (c) 2021-2023 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import { RuleTester } from 'eslint';
+
+import rule from './regular-expression-comment';
+
+// file.only
```
Comment (bobwieler) [inline src/regular-expression-comment.spec.ts:13]: should this be here?

**Review by bobwieler — CHANGES_REQUESTED** [review]
just a minor comment/question

File: src/regular-expression-comment.spec.ts (hunk 2 above)
Comment (jpolavar) [reply src/regular-expression-comment.spec.ts:13]: will remove it

**Review by carlansley — APPROVED** [review]

**Review by bobwieler — APPROVED** [review]
lgtm

**Pull request merged**